src/TorchSharp/TorchVision/Ops.cs

```diff
@@ -30,5 +33,48 @@ public static Tensor sigmoid_focal_loss(Tensor inputs, Tensor targets, float alp
                 return loss.MoveToOuterDisposeScope();
             }
         }
+
+        public static Tensor nme(Tensor boxes, Tensor scores, double iou_threshold = 0.5)
+        {
+            using (var nmsScope = torch.NewDisposeScope()) {
+                var x1 = boxes[.., 0];
+                var y1 = boxes[.., 1];
+                var x2 = boxes[.., 2];
+                var y2 = boxes[.., 3];
+                var areas = (x2 - x1) * (y2 - y1);
+                var (_, _order) = scores.sort(0, descending: true);
+                var order = _order[..];
+
+                var keep = new List<long>();
+                while (order.numel() > 0) {
+                    long i;
+                    if (order.numel() == 1) {
+                        i = order.item<long>();
+                        keep.Add(i);
+                        break;
+                    } else {
+                        i = order[0].item<long>();
+                        keep.Add(i);
+                    }
+
+                    var xx1 = x1[order[1..]].clamp(min: x1[i]);
```

Comment:
Same here -- can't use ranges in implementation code.

Reply:
changed it to tensor indices, like this: torch.arange(..).

Reply:
TensorIndex.Slice(1,null) may be faster.